src/Game.tsx

```diff
@@ -3,34 +3,109 @@ import { useState, useEffect } from 'react';
 import { Ball } from './Ball';
 import { PlayerCard } from './PlayerCard';
 
+type TimeoutResult = ReturnType<typeof setTimeout> | null;
+
 type Props = {
-  pokemons: [string, string];
-};
-type Coords = {
-  x: number;
-  y: number;
+  pokemonOne: string;
+  pokemonTwo: string;
 };
-export function Game({ pokemons }: Props): JSX.Element {
-  const [ballCoords, setBallCoods] = useState<Coords>({ x: 50, y: 50 });
-  const [player1Coord, setPlayer1Coord] = useState(0);
-  const [player2Coord, setPlayer2Coord] = useState(100);
-
-  function changePlayerCoords(e: KeyboardEvent): void {
-    console.log(e.key);
-    // Change state here
-    setBallCoods({ x: 100, y: 100 });
-    setPlayer1Coord(50);
-    setPlayer2Coord(50);
-  }
+type BallCoords = [number, number];
+
+const PLAYER_COORD_INTERVAL = 40;
+const PLAYER_COORD_STEP = 5;
+
+function calculateCoordMinusStep(oldCoord: number): number {
+  if (oldCoord > 0) return oldCoord - PLAYER_COORD_STEP;
+  return oldCoord;
+}
+
+function calculateCoordPlusStep(oldCoord: number): number {
+  if (oldCoord < 100) return oldCoord + PLAYER_COORD_STEP;
+  return oldCoord;
+}
+export function Game({ pokemonOne, pokemonTwo }: Props): JSX.Element {
+  const [ballCoords] = useState<BallCoords>([50, 50]);
+  const [ballCoordY, ballCoordX] = ballCoords;
+  const [player1Coord, setPlayer1Coord] = useState(50);
+  const [player2Coord, setPlayer2Coord] = useState(50);
+  const [isGameStarted, setIsGameStarted] = useState(false);
+
   useEffect(() => {
-    document.addEventListener('keydown', changePlayerCoords);
+    let firstPlayerInt: TimeoutResult = null;
+    let secondPlayerInt: TimeoutResult = null;
+
+    const handleKeyUp = (e?: KeyboardEvent) => {
+      if (!e) {
+        if (firstPlayerInt) clearInterval(firstPlayerInt);
+        firstPlayerInt = null;
+        if (secondPlayerInt) clearInterval(secondPlayerInt);
+        secondPlayerInt = null;
+      } else if (['w', 'W', 's', 'S'].includes(e.key)) {
+        if (firstPlayerInt) clearInterval(firstPlayerInt);
+        firstPlayerInt = null;
+      } else if (['ArrowUp', 'ArrowDown'].includes(e.key)) {
+        if (secondPlayerInt) clearInterval(secondPlayerInt);
+        secondPlayerInt = null;
+      }
+    };
+
+    const handleKeyDown = (e: KeyboardEvent) => {
+      switch (e.key) {
+        case 'w':
+        case 'W':
+          firstPlayerInt =
+            firstPlayerInt ||
+            setInterval(() => {
+              setPlayer1Coord((player1Coord) =>
+                calculateCoordMinusStep(player1Coord)
+              );
```

Comment:
This code creates unnecessary closure. Simplify by directly passing `calculateCoordMinusStep` function.